Developers SBAT know that the oldest oracle report used by CP-DOTO hasn't expired #299

Holymir opened the pull request:
### Description

Developers SBAT know that the oldest oracle report used by CP-DOTO hasn't expired #299

### Tested

unit test

Commit:
isReportActive function implemented, unit tests for isReportActive, isReportActive added in MockSortedOracles fot testing purposes

yerdua commented on packages/protocol/contracts/stability/SortedOracles.sol:
```diff
@@ -118,6 +118,21 @@ contract SortedOracles is ISortedOracles, Ownable, Initializable {
+   * @notice Check if last report is expired.
+   * @param token The address of the token for which the Celo Gold exchange rate is being reported.
+   */
+  function isReportActive(address token) external view returns (bool) {
```
The purpose of this function would be more clear with a name that includes the word "expired", like: `isOldestReportExpired`, `expiredReportsExist`, `oldestReportExpired`, etc

Unless we're planning to introduce other qualities that could make a report "active" or "inactive", I think we should stick with "expired" as the adjective here.

asaj replied:
+1 to `isOldestReportExpired`

yerdua commented on packages/protocol/contracts/stability/SortedOracles.sol:
```diff
@@ -118,6 +118,21 @@ contract SortedOracles is ISortedOracles, Ownable, Initializable {
+    if (now.sub(timestamp) >= reportExpirySeconds) {
+      return false;
+    }
+    return true;
```
Since this is the same as the check that's used in `removeExpiredReports`, could this reasonably be used inside that function?

Are there any complexities with `external` vs `public` that make code duplication preferable?

asaj replied:
+1, `removeExpiredReports` can remove the oldest reports until `isOldestReportExpired` returns false

Commit:
function isReportActive renamed to isOldestReportExpired and reused by removeExpiredReports,
isOldestReportExpired now returns bool isReportExpired and address of the current report,

asaj commented on packages/protocol/test/stability/sortedoracles.ts:
```diff
@@ -158,6 +159,41 @@ contract('SortedOracles', (accounts: string[]) => {
+    })
+
+    it('should return true if there are no reports', async () => {
+      let isReportExpired = await sortedOracles.isOldestReportExpired(aToken)
```
`const`

Commit:
Merge branch 'master' into limechain/299

Commit:
Merge branch 'master' into limechain/299
